test/performance/range/gap_decorator_random_benchmark.cpp

```diff
@@ -0,0 +1,283 @@
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+typedef long double time_type;
+
+/* Helper function to sample gap length for each ungapped sequence position.
+ *
+ * Parameter:
+ * size_type    size type of underlying sequence range
+ *
+ * Arguments:
+ * gap_vector   reference to empty vector storing the sampled gap lengths
+ * size         final (aligned) sequence size
+ */
+template<typename size_type>
+void sample(std::vector<size_type> & gap_vector, size_type size)
+{
+    std::default_random_engine generator;
+    generator.seed(std::chrono::system_clock::now().time_since_epoch().count());
+    std::uniform_real_distribution<double> uni(0.0, 1.0);
+    std::array<double,10> cumsum = {0.6395, 0.8263, 0.8871, 0.9257, 0.9544, 0.9709, 0.9813, 0.9890, 0.9955, 1.0000};
+    for (size_type i = 0; i < size; ++i){
+        double y = uni(generator);
+        gap_vector[i] = y;
+        auto it = std::find_if(cumsum.begin(), cumsum.end(), [y](double i){return y <= i;});
+        gap_vector[i] = it - cumsum.begin();
+    }
+}
+
+/* Helper function to adjust the ungapped sequence length w.r.t. sampled gaps s.t.
+ * the gapped sequence length does not exceed the targeted length.
+ *
+ * Parameters:
+ * size_type        size type of the sequence
+ * sequence_type    ungapped sequence type
+ *
+ * Arguments:
+ * gaps         reference to the gap vector
+ * seq          reference to ungapped sequence
+ * seq_len      final sequence length
+ */
+template<typename size_type, typename sequence_type>
+void resize(std::vector<size_type> & gaps, sequence_type & seq, unsigned int seq_len)
+{
+    size_type letter_acc = 0;
+    size_type gap_pos = 0;
+    size_type gap_acc = 0;
+
+    while (gap_pos < gaps.size() && gap_acc + letter_acc < seq_len)
+    {
+        if (!gaps[gap_pos])
+            ++letter_acc;
+        else
+        {
+            if (letter_acc + gap_acc + gaps[gap_pos] > seq_len)
+            {
+                gaps[gap_pos] = seq_len - gap_acc - letter_acc;
+                gap_acc += gaps[gap_pos];
+                ++gap_pos;
+                break;
+            }
+            else
+                gap_acc += gaps[gap_pos];
+        }
+        ++gap_pos;
+    }
+    seq.resize(std::max<size_type>(1, letter_acc));  // resize ungapped sequence
+    gaps.resize(gap_pos);      // trim sampled gap vector
+}
+
+/* Helper function to prepare a gapped sequence for the benchmark (case gap_flag=true)
+ *
+ * Parameters:
+ * size_type        size type of the sequence
+ * gap_decorator_t  gap decorator type, e.g. gap_decorator_anchor_set
+ *
+ * Arguments:
+ * gaps             reference to gap vector
+ * gap_decorator    reference to gap decorator
+ */
+template<typename size_type, typename gap_decorator_t>
+void insert_gaps(std::vector<size_type> & gaps, gap_decorator_t & gap_decorator)
```

Comment:
```suggestion
void insert_gaps(std::vector<typename gap_decorator_t::size_type> & gaps, gap_decorator_t & gap_decorator)
```
Then you don't need the first template paramter